sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/DeadletterQueueSample.java

```diff
@@ -0,0 +1,222 @@
+            message.setTimeToLive(Duration.ofMinutes(2));
+            System.out.printf("\tMessage sending: Id = %s\n", message.getMessageId());
+            senderClient.sendMessage(message);
+            System.out.printf("\tMessage acknowledged: Id = %s\n", message.getMessageId());
+        }
+    }
+
+    /**
+     * Receive {@link ServiceBusMessage messages} and return {@link ServiceBusMessage messages} back to the queue.
+     * When the time to life of the {@link ServiceBusMessage messages} expires,
+     * the {@link ServiceBusMessage messages} will be dumped as dead letters into the dead letter queue.
+     * We can receive these {@link ServiceBusMessage messages} from the dead letter queue.
+     *
+     * @Param connectionString Service Bus Connection String
+     * @Param queueName Queue Name
+     * @throws InterruptedException If the program is unable to sleep while waiting for the receive to complete.
+     */
+    void deadLetterByExceedingMaxDelivery(String connectionString, String queueName) throws InterruptedException {
+        ServiceBusReceiverAsyncClient receiverAsyncClient
+            = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .receiver()
+            .queueName(queueName)
+            .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
+            .buildAsyncClient();
+        receiverAsyncClient.receiveMessages().subscribe(receiveMessage -> {
+            System.out.printf("Picked up message; DeliveryCount %d\n", receiveMessage.getDeliveryCount());
+            // return message back to the queue
+            receiverAsyncClient.abandon(receiveMessage);
+        });
+        Thread.sleep(10000);
+        receiverAsyncClient.close();
+
+        Thread.sleep(120000);
+
+        ServiceBusReceiverAsyncClient deadletterReceiverAsyncClient
+            = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .receiver()
+            .queueName(queueName.concat("/$deadletterqueue"))
+            .receiveMode(ServiceBusReceiveMode.PEEK_LOCK)
+            .buildAsyncClient();
+        deadletterReceiverAsyncClient.receiveMessages().subscribe(receiveMessage -> {
+            System.out.printf("\nDeadletter message:\n");
+            receiveMessage.getApplicationProperties().keySet().forEach(key -> System.out.printf("\t%s=%s\n", key, receiveMessage.getApplicationProperties().get(key)));
+            deadletterReceiverAsyncClient.complete(receiveMessage);
+        });
+        Thread.sleep(10000);
+        deadletterReceiverAsyncClient.close();
+    }
+
+    /**
+     * Receive {@link ServiceBusMessage messages} and transfer to the dead letter queue as a dead letter.
+     *
+     * @Param connectionString Service Bus Connection String
+     * @Param queueName Queue Name
+     */
+    void receiveAndDeadletterMessagesAsync(String connectionString, String queueName) {
+        ServiceBusReceiverAsyncClient receiverAsyncClient
```

Comment:
Same comment as above, please change this to Mono<Void> instead of void.

Reply:
According to your Comment, fixed in new version.